orangecontrib/text/twitter.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import tweepy
+from threading import Lock
+
+from orangecontrib.text.corpus import Corpus
+
+from Orange import data
+
+__all__ = ['Credentials', 'TwitterAPI']
+
+
+class Credentials:
+    """Twitter api credentials."""
+
+    def __init__(self, consumer_key, consumer_secret, callback=None):
+        self.consumer_key = consumer_key
+        self.consumer_secret = consumer_secret
+        self.callback = callback
+        self.auth = tweepy.OAuthHandler(self.consumer_key, self.consumer_secret,
+                                        callback=self.callback)
+
+    def check_credentials(self):
+        try:
+            self.auth.get_authorization_url()
+        except tweepy.TweepError:
+            return False
+
+        return True
+
+    def construct_api(self):
+        return tweepy.API(self.auth)
+
+    def __eq__(self, other):
+        return self.consumer_key == other.consumer_key \
+               and self.consumer_secret == other.consumer_secret
+
+
+class TwitterAPI:
+    """Fetch tweets from the Tweeter API"""
+
+    string_fields = [
+        'text', 'created_at', 'lang'
+    ]
+
+    continuous_fields = [
+        'favorite_count', 'retweet_count'
+    ]
+
+    discrete_fields = [
+
+    ]
+    supported_fields = string_fields + continuous_fields + discrete_fields
+
+    def __init__(self, credentials):
+        """
+
+        Args:
+            credentials (Credentials):
+        """
+        self.key = credentials
+        self.api = credentials.construct_api()
+        self.stream = tweepy.Stream(auth=self.api.auth, listener=Listener(self))
+        self.statuses_lock = Lock()
+
+        self.container = []
+
+    # TODO: add arguments
+    def search(self, *, word_list, max_tweets, lang=None):
+        if not word_list:
+            raise ValueError("Word list can't be empty.")
+
+        query = ' OR '.join(word_list)
+
+        for status in tweepy.Cursor(self.api.search, q=query, lang=lang).items(max_tweets):
+            self.add_status(status)
+
+    def listen(self, *, word_list):
+        self.stream.filter(track=word_list, async=True)
+
+    def add_status(self, status):
+        status_record = {field: getattr(status, field)
+                         for field in self.supported_fields}
+        self.statuses_lock.acquire()
```

Comment:
OK, I agree we keep it _if_ something will be asynchronous or in the background.